RDMA.md

```diff
@@ -0,0 +1,174 @@
+  RDMA [WRITE][WRITE][WRITE]...[WRITE WITH IMM], the length of total buffer is described by
+  immediate data(unsigned int 32).
+
+
+### Maximum WQE(s) of RDMA
+Currently no specific restriction is defined in this protocol. Recommended WQEs is 1024.
+Flow control for WQE MAY be defined/implemented in the future.
+
+
+### The workflow of this protocol
+```
+                                                                    valkey-server
+                                                                    listen RDMA port
+   valkey-client
+                -------------------RDMA connect-------------------->
+                                                                    accept connection
+                <--------------- Establish RDMA --------------------
+
+                --------Get server feature [@IBV_WR_SEND] --------->
+
+                --------Set client feature [@IBV_WR_SEND] --------->
+                                                                    setup RX buffer
+                <---- Register transfer memory [@IBV_WR_SEND] ------
+[@ibv_post_recv]
+setup TX buffer
+                ----- Register transfer memory [@IBV_WR_SEND] ----->
+                                                                    [@ibv_post_recv]
+                                                                    setup TX buffer
+                -- Valkey commands [@IBV_WR_RDMA_WRITE_WITH_IMM] -->
+                <- Valkey response [@IBV_WR_RDMA_WRITE_WITH_IMM] ---
+                                  .......
+                -- Valkey commands [@IBV_WR_RDMA_WRITE_WITH_IMM] -->
+                <- Valkey response [@IBV_WR_RDMA_WRITE_WITH_IMM] ---
+                                  .......
+
+
+RX is full
+                ----- Register transfer memory [@IBV_WR_SEND] ----->
+                                                                    [@ibv_post_recv]
+                                                                    setup TX buffer
+                <- Valkey response [@IBV_WR_RDMA_WRITE_WITH_IMM] ---
+                                  .......
+
+                                                                    RX is full
+                <---- Register transfer memory [@IBV_WR_SEND] ------
+[@ibv_post_recv]
+setup TX buffer
+                -- Valkey commands [@IBV_WR_RDMA_WRITE_WITH_IMM] -->
+                <- Valkey response [@IBV_WR_RDMA_WRITE_WITH_IMM] ---
+                                  .......
+
+                -------------------RDMA disconnect----------------->
+                <------------------RDMA disconnect------------------
+```
+
+
+## Event handling
+There is no POLLOUT event of RDMA comp channel:
+   1, if TX is not full, it's always writable.
+   2, if TX is full, should wait a 'RegisterLocalAddr' message to refresh
+      'TX buffer'.
+
+To-Do List
+----------
+- [ ] hiredis
+- [ ] rdma client & benchmark
+- [ ] POLLOUT event emulation for hiredis
+- [ ] auto-test suite is not implemented currently
```

Comment:
This can be in the PR description or create github issues for them. It doesn't belong in the code repo.

Reply:
> This can be in the PR description or create github issues for them. It doesn't belong in the code repo.

OK.